tests/unit/HeadlessLayerImplicitAnimationTests.swift

```diff
@@ -29,52 +29,101 @@ import MotionAnimator
 class HeadlessLayerImplicitAnimationTests: XCTestCase {
 
   var window: UIWindow!
-  var layer: CALayer!
   override func setUp() {
     super.setUp()
 
     window = UIWindow()
     window.makeKeyAndVisible()
-
-    layer = CALayer()
-
-    window.layer.addSublayer(layer)
-
-    // Connect our layers to the render server.
-    CATransaction.flush()
   }
 
   override func tearDown() {
-    layer = nil
     window = nil
 
     super.tearDown()
   }
 
-  func testUnFlushedLayerDoesNotImplicitlyAnimateInCATransaction() {
+  func testViewDoesImplicitlyAnimateInUIViewAnimateBlock() {
+    let view = UIView()
+    UIView.animate(withDuration: CATransaction.animationDuration() + 0.1) {
+      view.alpha = 0.5
+    }
+
+    XCTAssert(view.layer.animation(forKey: "opacity") is CABasicAnimation)
+    if let animation = view.layer.animation(forKey: "opacity") as? CABasicAnimation {
+      XCTAssertEqual(animation.keyPath, "opacity")
+      XCTAssertEqual(animation.duration, CATransaction.animationDuration() + 0.1)
+    }
+  }
+
+  func testViewLayerDoesImplicitlyAnimateInUIViewAnimateBlock() {
+    let view = UIView()
+    UIView.animate(withDuration: CATransaction.animationDuration() + 0.1) {
+      view.layer.opacity = 0.5
+    }
+
+    XCTAssert(view.layer.animation(forKey: "opacity") is CABasicAnimation)
+    if let animation = view.layer.animation(forKey: "opacity") as? CABasicAnimation {
+      XCTAssertEqual(animation.keyPath, "opacity")
+      XCTAssertEqual(animation.duration, CATransaction.animationDuration() + 0.1)
+    }
+  }
+
+  func testViewDoesNotImplicitlyAnimate() {
+    let view = UIView()
+    view.alpha = 0.5
+
+    XCTAssertNil(view.layer.animationKeys())
+  }
+
+  func testViewLayerDoesNotImplicitlyAnimate() {
+    let view = UIView()
+    view.layer.opacity = 0.5
+
+    XCTAssertNil(view.layer.animationKeys())
+  }
+
+  func testUnFlushedLayerDoesNotImplicitlyAnimate() {
     let unflushedLayer = CALayer()
-    CATransaction.begin()
-    CATransaction.setAnimationDuration(0.5)
     unflushedLayer.opacity = 0.5
-    CATransaction.commit()
 
     XCTAssertNil(unflushedLayer.animationKeys())
   }
 
-  func testDoesImplicitlyAnimateInCATransaction() {
-    CATransaction.begin()
-    CATransaction.setAnimationDuration(0.5)
+  func testDoesImplicitlyAnimate() {
```

Comment:
Nit/Naming: "**What** is implicitly animating?"  `testUnhostedLayerInWindowDoesImplicitlyAnimate`?